src/data_conn.c

```diff
@@ -264,21 +264,34 @@ handle_data_conn_error(replica_t *r)
 	int fd, data_eventfd, mgmt_eventfd2, epollfd;
 	spec_t *spec;
 	replica_t *r1 = NULL;
+	int found_in_list = 0;
 
 	if (r->iofd == -1) {
-		REPLICA_ERRLOG("repl %s %d %p\n", r->ip, r->port, r);
+		REPLICA_ERRLOG("repl %s %d %p data_conn error\n", r->ip, r->port, r);
 		return -1;
 	}
 
 	MTX_LOCK(&r->spec->rq_mtx);
 	spec = r->spec;
 
 	TAILQ_FOREACH(r1, &(spec->rq), r_next)
-		if (r1 == r)
+		if (r1 == r) {
+			assert(r->quorum == 1);
+			found_in_list = 1;
 			break;
+		}
+
+	if (r1 == NULL)
+		TAILQ_FOREACH(r1, &(spec->non_quorum_rq), r_non_quorum_next)
+			if (r1 == r) {
+				assert(r->quorum == 0);
+				found_in_list = 2;
+				break;
+			}
 
 	if (r1 == NULL) {
-		REPLICA_ERRLOG("replica %s %d not part of rqlist..\n",
+		MTX_LOCK(&r->r_mtx);
+		REPLICA_ERRLOG("replica %s %d not part of rq and non_rq list..\n",
```

Comment:
move this print out of the lock, and below comment need to be updated by saying that 'replica hasn't been added to spec's rqlist and non_rq_list'

Reply:
Done sir.